src/main/java/seedu/guest/model/guest/Request.java

```diff
@@ -0,0 +1,68 @@
+package seedu.guest.model.guest;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.guest.commons.util.AppUtil.checkArgument;
+
+/**
+ * The class Request.
+ */
+public class Request {
+    public static final String MESSAGE_CONSTRAINTS = "The request should not be null";
+    public final String value;
+
+    /**
+     * The constructor with no specific request.
+     */
+    public Request() {
+        value = "Nothing";
+    }
+
+    /**
+     * The constructor with specific Request object.
+     * @param str the specific Request
+     */
+    public Request(String str) {
+        requireNonNull(str);
+        checkArgument(isValidRequest(str), MESSAGE_CONSTRAINTS);
+        value = str;
+    }
+
+    /**
+     * To check whether the Request is valid.
+     * @param inputRequest
+     * @return boolean
+     */
+    public static boolean isValidRequest(String inputRequest) {
+        return (!inputRequest.equals(""));
```

Comment:
Why do we not allow ""?

Reply:
cuz it will be null. My thoughts is that either there are no request or there must be at least a word